Source/AliveLibAE/Abe.cpp

```diff
@@ -3267,49 +3271,55 @@ void Abe::State_0_Idle_44EEB0()
 
     if (!sInputObject_5BD4E0.isPressed(sInputKey_Up_5550D8) || handleDoActionOrThrow)
     {
-        // TODO: Clean up the logic of these 2 statements
-        if (!(sInputKey_ThrowItem_5550F4 & held) || field_106_current_motion != eAbeStates::State_0_Idle_44EEB0)
+        if ((sInputKey_ThrowItem_5550F4 & held) &&
+            field_106_current_motion == eAbeStates::State_0_Idle_44EEB0)
         {
-            if (held & sInputKey_DoAction_5550E4) // not throwing, maybe pressing up and pressing action, so do action
+            if (field_1A2_throwable_count > 0 || gInfiniteGrenades_5C1BDE)
             {
-                field_106_current_motion = HandleDoAction_455BD0();
-            }
-        }
-        else if (field_1A2_throwable_count > 0 || gInfiniteGrenades_5C1BDE)
-        {
-            field_158_throwable_id = Make_Throwable_49AF30(
-                field_B8_xpos,
-                field_BC_ypos - FP_FromInteger(40),
-                0)->field_8_object_id;
+                field_158_throwable_id = Make_Throwable_49AF30(
+                    field_B8_xpos,
+                    field_BC_ypos - FP_FromInteger(40),
+                    0)->field_8_object_id;
 
-            if (!bThrowableIndicatorExists_5C112C)
-            {
-                ThrowableTotalIndicator* pThrowable = ae_new<ThrowableTotalIndicator>();
-                if (pThrowable)
+                if (!bThrowableIndicatorExists_5C112C)
                 {
-                    const FP xOffSet = ((field_20_animation.field_4_flags.Get(AnimFlags::eBit5_FlipX)) ? FP_FromInteger(15) : FP_FromInteger(-15)) * field_CC_sprite_scale;
-
-                    pThrowable->ctor_431CB0(
-                        field_B8_xpos + xOffSet,
-                        field_BC_ypos + (field_CC_sprite_scale * FP_FromInteger(-50)),
-                        field_20_animation.field_C_render_layer,
-                        field_20_animation.field_14_scale,
-                        field_1A2_throwable_count,
-                        TRUE);
+                    ThrowableTotalIndicator* pThrowable = ae_new<ThrowableTotalIndicator>();
```

Comment:
ditto for auto?

Reply:
Again, not my doing, just so we're in the clear :tape: